pkg/mount-manager/safe-mounter_windows.go

```diff
@@ -0,0 +1,221 @@
+// +build windows
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mountmanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	diskv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/api/disk/v1alpha1"
+	diskclientv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/groups/disk/v1alpha1"
+
+	fsv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/api/filesystem/v1alpha1"
+	fsclientv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/groups/filesystem/v1alpha1"
+
+	volumev1alpha1 "github.com/kubernetes-csi/csi-proxy/client/api/volume/v1alpha1"
+	volumeclientv1alpha1 "github.com/kubernetes-csi/csi-proxy/client/groups/volume/v1alpha1"
+
+	utilexec "k8s.io/utils/exec"
+	"k8s.io/utils/mount"
+)
+
+var _ mount.Interface = &CSIProxyMounter{}
+
+type CSIProxyMounter struct {
+	FsClient     *fsclientv1alpha1.Client
+	DiskClient   *diskclientv1alpha1.Client
+	VolumeClient *volumeclientv1alpha1.Client
+}
+
+func NewCSIProxyMounter() (*CSIProxyMounter, error) {
+	fsClient, err := fsclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	diskClient, err := diskclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	volumeClient, err := volumeclientv1alpha1.NewClient()
+	if err != nil {
+		return nil, err
+	}
+	return &CSIProxyMounter{
+		FsClient:     fsClient,
+		DiskClient:   diskClient,
+		VolumeClient: volumeClient,
+	}, nil
+}
+
+func NewSafeMounter() (*mount.SafeFormatAndMount, error) {
+	csiProxyMounter, err := NewCSIProxyMounter()
+	if err != nil {
+		return nil, err
+	}
+	return &mount.SafeFormatAndMount{
+		Interface: csiProxyMounter,
+		Exec:      utilexec.New(),
+	}, nil
+}
+
+func (mounter *CSIProxyMounter) Mount(source string, target string, fstype string, options []string) error {
+	// Mount is called after the format is done.
+	// TODO: Confirm that fstype is empty.
+	// Call the LinkPath CSI proxy from the source path to the target path
+	parentDir := filepath.Dir(target)
+	if _, err := os.Stat(target); !os.IsNotExist(err) {
+		os.RemoveAll(target)
```

Comment:
Generally we try to avoid calling RemoveAll because we had cases in the past where the volume was still mounted, and RemoveAll ended up deleting user data.

Reply:
fixed, it is removed because preparePublishPath did it